TESTS/netsocket/tls/tlssocket_bind_unopened.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "greentea-client/test_env.h"
+#include "mbed.h"
+#include "tls_tests.h"
+#include "TLSSocket.h"
+#include "unity/unity.h"
+#include "utest.h"
+
+using namespace utest::v1;
+
+void TLSSOCKET_BIND_UNOPENED()
+{
+#if MBED_CONF_NSAPI_SOCKET_STATS_ENABLE
+    int count = fetch_stats();
+    for (int j = 0; j < count; j++) {
+        TEST_ASSERT_EQUAL(SOCK_CLOSED,  tls_stats[j].state);
+    }
+#endif
+
+    TLSSocket *sock = new TLSSocket;
+    if (!sock) {
+        TEST_FAIL();
```

Comment:
TEST_FAIL won't abort execution so adding return-statement would be good here.

Reply:
Good point, but I removed the bind tests entirely. 
However, these same comments apply also to tcp tests.